File: 2025-2/sesion-de-clase-11/SoftProgNegocio/BOImpl/EmpleadoBOImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PUCP.SoftProg.Modelo;
using PUCP.SoftProg.Modelo.RRHH;
using PUCP.SoftProg.Negocio.BO;
using PUCP.SoftProg.Persistencia.DAO.RRHH;
using PUCP.SoftProg.Persistencia.DAOImpl.RRHH;

namespace PUCP.SoftProg.Negocio.BOImpl {
    public class EmpleadoBOImpl : IEmpleadoBO {
        private readonly IEmpleadoDAO empleadoDAO;

        public EmpleadoBOImpl() { 
            this.empleadoDAO = new EmpleadoDAOImpl();
        }

        public Empleado BuscarPorDni(string dni) {
            return this.empleadoDAO.BuscarPorDni(dni);
        }

        public void Eliminar(int id) {
            this.empleadoDAO.Eliminar(id);
        }

        public void Guardar(Empleado empleado, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.empleadoDAO.Crear(empleado);
            }
            else if (estado == Estado.Modificado) {
                this.empleadoDAO.Actualizar(empleado);
            }
        }

        public List<Empleado> Listar() {
            return this.empleadoDAO.LeerTodos();
        }

        public Empleado Obtener(int id) {
            return this.empleadoDAO.Leer(id);
        }
    }
}

File: 2025-2/sesion-de-clase-18/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/rrhh/EmpleadoBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl.rrhh;

import java.util.List;
import pe.edu.pucp.inf30.softprog.dao.rrhh.EmpleadoDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.rrhh.EmpleadoDAOImpl;
import pe.edu.pucp.inf30.softprog.modelo.Estado;
import pe.edu.pucp.inf30.softprog.modelo.rrhh.Empleado;
import pe.edu.pucp.inf30.softprog.bo.rrhh.EmpleadoBO;

/**
 *
 * @author eric
 */
public class EmpleadoBOImpl implements EmpleadoBO {

    private final EmpleadoDAO empleadoDao;
    
    public EmpleadoBOImpl() {
        this.empleadoDao = new EmpleadoDAOImpl();
    }
    
    @Override
    public List<Empleado> listar() {
        return this.empleadoDao.leerTodos();
    }

    @Override
    public Empleado obtener(int id) {
        return this.empleadoDao.leer(id);
    }

    @Override
    public void eliminar(int id) {
        this.empleadoDao.eliminar(id);
    }

    @Override
    public void guardar(Empleado modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            this.empleadoDao.crear(modelo);
        }
        else {
            this.empleadoDao.actualizar(modelo);
        }
    }

    @Override
    public Empleado buscarPorDni(String dni) {
        return this.empleadoDao.buscarPorDni(dni);
    }
}
